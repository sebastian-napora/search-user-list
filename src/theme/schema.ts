import { MEDIA_QURIES } from './mediaQueries';

import { TThemes } from './types';

const Theme = {
  data: 'main',
  id: 'T_001',
  name: 'Main',
  colors: {
    fonts: {
      primary: '#f1f1f1',
      white: '#ffffff' as const,
      grey: '#838383',
      greySecondary: '#a6a6a6'
    },
    white: '#ffffff' as const,
    body: '#FFFFFF',
    primary: '#490e4b',
    secondary: '#3b0044',
    greyPrimary: '#cfcfcf',
    greySecondary: '#e1dcdb',
    hoverColor: '#f5f5f5' as const,
    black: '#020000',
    text: '#000000',
    label: '#838383',
    button: {
      text: '#FFFFFF',
      background: '#000000'
    },
    link: {
      text: '#3b0044',
      opacity: 1
    }
  },
  mediaQueries: MEDIA_QURIES,
  font: {
    text: 'AauxPro OT'
  }
};

//changed in the future this is only example
export const THEMES: TThemes = {
  data: {
    main: Theme,
    otherTheme: Theme
  }
};
