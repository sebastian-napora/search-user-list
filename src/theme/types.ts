// eslint-disable-next-line import/named
import { TCss } from 'styled-components';

export type TThemes = {
  data: {
    main: TSingleTheme;
    otherTheme: TSingleTheme;
  };
};

export type TUseTheme = {
  theme: TSingleTheme;
  themeLoaded: boolean;
  setMode: (mode: TSingleTheme) => void;
  handleSetTheme: (localTheme: TThemes) => void;
};

export type TSingleTheme = {
  data: unknown;
  id: string;
  name: string;
  colors: {
    fonts: {
      primary: string;
      white: '#ffffff';
      grey: string;
      greySecondary: string;
    };
    body: string;
    text: string;
    white: '#ffffff';
    primary: string;
    secondary: string;
    black: string;
    greyPrimary: string;
    greySecondary: string;
    hoverColor: '#f5f5f5';
    button: {
      text: string;
      background: string;
    };
    link: {
      text: string;
      opacity: number;
    };
  };
  font: {
    text: string;
  };
  mediaQueries: {
    smartphones: string;
    smallTablet: string;
    tablet: string;
    laptop: string;
    desktop: string;
  };
};

export type TCssStylesByDeviceWidth = (
  typeDevice: 'smartphones' | 'smallTablet' | 'tablet' | 'laptop' | 'desktop',
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  props: any
) => TCss;
