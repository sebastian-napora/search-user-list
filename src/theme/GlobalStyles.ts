import { createGlobalStyle } from 'styled-components';

import { TSingleTheme } from './types';

export const GlobalStyles = createGlobalStyle<{ theme: TSingleTheme }>`
  body {
    background: ${({ theme }) => theme.colors.body};
    color: ${({ theme }) => theme.colors.text};
    font-family: ${({ theme }) => theme.font.text};
    height: 100vh;
    margin: 0 !important;
  }

  a {
    color: ${({ theme }) => theme.colors.link.text};
    cursor: pointer;
  }

  button {
    border: 0;
    display: inline-block;
    padding: 12px 24px;
    font-size: 14px;
    border-radius: 4px;
    margin-top: 5px;
    cursor: pointer;
    background-color: #1064EA;
    color: #ffffff;
    font-family: ${({ theme }) => theme.font};
  }

  button.btn {
    background-color: ${({ theme }) => theme.colors.button.background};
    color: ${({ theme }) => theme.colors.button.text};
  }
`;
