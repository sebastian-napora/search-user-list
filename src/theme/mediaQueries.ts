import { css } from 'styled-components';
import { TCssStylesByDeviceWidth } from './types';

export const SIZE = {
  smartphoneMin: 320,
  smartphoneMax: 480,
  smallTablet: 767,
  tabletMin: 768,
  tabletMax: 1024,
  laptopMin: 1025,
  laptopMax: 1280,
  desktop: 1281
};

export const MEDIA_QURIES = {
  /*
    ##Device = Most of the Smartphones Mobiles (Portrait)
    ##Screen = B/w 320px to 479px
  */
  smartphones: `(min-width: ${SIZE.smartphoneMin}px)`,
  /*
    ##Device = Low Resolution Tablets, Mobiles (Landscape)
    ##Screen = B/w 481px to 767px
  */
  smallTablet: `(min-width: ${SIZE.smartphoneMax + 1}px)`,
  /*
    ##Device = Tablets
    ##Screen = B/w 768px to 1024px
  */
  tablet: `(min-width: ${SIZE.tabletMin}px)`,
  /*
    ##Device = Laptops, Desktops
    ##Screen = B/w 1025px to 1280px
  */
  laptop: `(min-width: ${SIZE.laptopMin}px)`,
  /*
    ##Device = Desktops
    ##Screen = 1281px to higher resolution desktops
  */
  desktop: `(min-width: ${SIZE.desktop}px)`
};

export const cssStylesByDeviceWidth: TCssStylesByDeviceWidth = (typeDevice, props) => css`
  @media ${MEDIA_QURIES[typeDevice]} {
    ${props}
  }
`;
