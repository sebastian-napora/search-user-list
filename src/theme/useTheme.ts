import { useEffect, useState } from 'react';

import { THEMES } from './schema';

import { TSingleTheme, TThemes, TUseTheme } from './types';

export const useTheme = (): TUseTheme => {
  const allThemes = THEMES;
  const [theme, setTheme] = useState<TSingleTheme>(allThemes.data.main);
  const [themeLoaded, setThemeLoaded] = useState<boolean>(false);

  const setMode = (mode: TSingleTheme): void => setTheme(mode);

  const handleSetTheme = (localThemes: TThemes): void => {
    localThemes && setTheme(localThemes.data.main);
    setThemeLoaded(true);
  };

  useEffect(() => handleSetTheme(THEMES), []);

  return {
    theme,
    themeLoaded,
    setMode,
    handleSetTheme
  };
};
