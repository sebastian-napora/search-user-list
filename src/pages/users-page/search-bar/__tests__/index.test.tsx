import React from 'react';
import { ThemeProvider } from 'styled-components';

import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { THEMES } from 'src/theme/schema';

import SearchBar from '../index';

afterEach(cleanup);

describe('SearchBar tests', () => {
  describe('1. Checked render children component SearchBar', () => {
    it('1.1. Should render children component well SearchBar', () => {
      const { container, unmount } = render(
        <ThemeProvider {...{ theme: THEMES.data.main }}>
          <SearchBar onChange={() => null} />
        </ThemeProvider>
      );

      expect(container).toMatchSnapshot();

      unmount();
    });
  });
});
