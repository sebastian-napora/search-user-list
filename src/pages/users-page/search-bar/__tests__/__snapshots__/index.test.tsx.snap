// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar tests checked render children component SearchBar should render children component well SearchBar 1`] = `
<div>
  <input
    class="sc-bdvvtL enLUtv"
    placeholder="Search by user name..."
    type="text"
  />
</div>
`;
